docs/src/rules/no-invalid-this.md

```diff
@@ -284,6 +284,62 @@ obj.Foo = function Foo() {
 
 :::
 
+This rule additionally supports TypeScript type syntax.
+
+Examples of **incorrect** TypeScript code for this rule:
+
+:::incorrect
+
+```ts
+/*eslint no-invalid-this: "error"*/
+
+interface SomeType {
+    prop: string;
+}
+
+function foo(bar: string) {
+    this.prop;
+    console.log(bar)
+}
+
+/** @this Obj */
+foo(function() {
+    console.log(this);
+    z(x => console.log(x, this));
+});
+
+```
+
+:::
+
+Examples of **correct** TypeScript code for this rule:
+
+:::correct
+
+```ts
+/*eslint no-invalid-this: "error"*/
+
+interface SomeType {
+    prop: string;
+}
+
+function foo(this: SomeType) {
+    this.prop;
+}
+
+class A {
+    a = 5;
+    b = this.a;
+    accessor c = this.a;
+}
+```
+
+:::
+
+**Note:** The code problem checked by this ESLint rule is automatically checked by the TypeScript compiler. Thus, it is not recommended to turn on this rule in new TypeScript projects. You only need to enable this rule if you prefer the ESLint error messages over the TypeScript compiler error messages or using it to lint JavaScript code.
```

Comment:
Let's move this note under `When not to use it` section, also let's udpate the text.

```suggestion
The TypeScript compiler already checks for the issues addressed by this ESLint rule. Therefore, enabling this rule in new TypeScript projects is generally unnecessary. You should only enable it if you prefer ESLint's error messages over the TypeScript compiler's or if you're using it to lint JavaScript code.
```

Reply:
Done!